lib/masamune/actions/invoke_parallel.rb

```diff
@@ -35,13 +35,16 @@ module InvokeParallel
     end
 
     def invoke_parallel(*task_group)
-      opts = task_group.last.is_a?(Hash) ? task_group.pop.dup : {}
-      max_tasks = [opts.delete(:max_tasks), task_group.count].min
+      per_task_opts = task_group.last.is_a?(Array) ? task_group.pop.dup : [{}]
```

Comment:
@eatstoomuchjam I think it might be copy-pasta from a part of the code that was interacting with `Thor` `options` which are frozen - seems safe to remove